server/home/home-service/src/main/java/io/holoinsight/server/home/biz/common/GaeaSqlTaskUtil.java

```diff
@@ -306,75 +308,74 @@ public static Where buildFrontFilterWhere(List<Filter> whiteFilters, List<Filter
-    // List<ColumnCalExpr> exprs = translate.exprs;
-    //
-    // Transform transform = new Transform();
-    // List<Elect.TransformItem> transformItems = new ArrayList<>();
-    // if (!CollectionUtils.isEmpty(exprs)) {
-    // exprs.forEach(expr -> {
-    // Elect.TransformItem transformItem = new TransformItem();
-    // transformItem.setArg(expr.getParams());
-    // transformItem.setFunc(expr.getFunc());
-    // transformItems.add(transformItem);
-    // });
-    // }
-    //
-    // transform.setTransforms(transformItems);
-    // elect.setTransform(transform);
-    // }
-    //
-    // if (!StringUtils.isEmpty(rule.defaultValue)) {
-    // elect.setDefaultValue(rule.defaultValue);
-    // }
-    //
-    // Where.In in = new Where.In();
-    // in.setValues(t.values);
-    // in.setElect(elect);
-    // and.setIn(in);
-    //
-    // ands.add(and);
-    // });
-    //
+
+    Map<String, SplitCol> dimColMap = splitColMap.get(dimColType);
+    if (!CollectionUtils.isEmpty(collectMetric.afterFilters)
+        && !CollectionUtils.isEmpty(dimColMap)) {
+      for (AfterFilter afterFilter : collectMetric.afterFilters) {
+        if (!dimColMap.containsKey(afterFilter.getName()))
+          continue;
+        SplitCol dim = dimColMap.get(afterFilter.getName());
+
+        Rule rule = dim.rule;
+        Elect elect = new Elect();
+
+        switch (logParse.splitType) {
+          case leftRight:
+            Elect.LeftRight leftRight = new Elect.LeftRight();
+            leftRight.setLeft(rule.left);
+            leftRight.setLeftIndex(rule.leftIndex);
+            leftRight.setRight(rule.right);
+
+            elect.setType("leftRight");
+            elect.setLeftRight(leftRight);
+            break;
+
+          case separator:
+            elect.setType("refIndex");
+            elect.setRefIndex(new RefIndex());
+            elect.getRefIndex().setIndex(rule.pos);
+            break;
+          case regexp:
+            if (StringUtils.isNotEmpty(rule.regexpName)) {
+              elect.setType("refName");
+              elect.setRefName(new RefName());
+              elect.getRefName().setName(rule.regexpName);
+            } else if (rule.pos != null) {
+              elect.setType("refIndex");
+              elect.setRefIndex(new RefIndex());
+              elect.getRefIndex().setIndex(rule.pos);
+            }
+            break;
+          default:
+            break;
+        }
+
+        Where and = new Where();
+        Where.In in = new In();
+
+        switch (afterFilter.filterType) {
+          case IN:
+            in.setElect(elect);
+            in.setValues(afterFilter.getValues());
+            and.setIn(in);
+            break;
+          case NOT_IN:
+            Where not = new Where();
+            in.setValues(afterFilter.getValues());
+            in.setElect(elect);
+            not.setIn(in);
+            and.setNot(not);
+            break;
+          default:
+            break;
```

Comment:
Should this line throws an exception if meet unknown filterType ?

Reply:
The type here is fixed by the front end, and no exception scenario will occur